Re factor ceph_integration module

shtripat opened the pull request.

Commit:
WIP - Refactor ceph_integration module

tendrl-spec: refactor_common_code_and_move
tendrl-bug-id: Tendrl/ceph_integration#59

Signed-off-by: Shubhendu <shtripat@redhat.com>

Commit:
Added/modified test cases as part of refactor

tendrl-spec: refactor_common_code_and_move
tendrl-bug-id: Tendrl/ceph_integration#59

Signed-off-by: Shubhendu <shtripat@redhat.com>

cybernth commented on tendrl/ceph_integration/flows/create_pool.py:
```diff
@@ -1,6 +1,16 @@
-class CreatePool(Flow):
+class CreatePool(BaseFlow):
     def run(self):
+        self.parameters.update(
```
Do we really need to pass this around? Instead directly call these from atoms.

shtripat replied:
Thats correct. we can import crud module and use the functions as is. Will change the logic accordingly.

cybernth commented on tendrl/ceph_integration/flows/delete_pool.py:
```diff
@@ -1,6 +1,16 @@
-class DeletePool(Flow):
+class DeletePool(BaseFlow):
     def run(self):
+        self.parameters.update(
```
Same as above

shtripat replied:
Done

cybernth commented on tendrl/ceph_integration/manager/manager.py:
```diff
@@ -114,46 +108,17 @@ def delete_cluster(self, fs_id):
-        self.persister.join()
-        # self.eventer.join()
-        # self.servers.join()
+        super(self).join()
```
Should it be super(CephIntegrationManager, self).join() ?

shtripat replied:
Actually in `__init__()` we use class name and other functions `self` is used

cybernth replied:
okay

cybernth commented on tendrl/ceph_integration/manager/manager.py:
```diff
@@ -204,7 +157,7 @@ def main():
                 cluster_id = sys.argv[2]
                 utils.set_tendrl_context(cluster_id)
```
Does it create any issues during restart?

shtripat replied:
I am not sure at the moment and as we discussed earlier ideally we should be starting bridges as well as services and not child processes of the node agent. We can do this as part of a separate PR preferably as it applies to both gluster and ceph integration modules.

cybernth replied:
okay, do we have a git issue to track this?

shtripat replied:
https://github.com/Tendrl/node_agent/issues/116 introduces starting of integration processes as services and not as command line process.

cybernth commented on tendrl/ceph_integration/persistence/persister.py:
```diff
@@ -115,17 +52,85 @@ def update_tendrl_definitions(self, definition):
-        while not self._complete.is_set():
-            gevent.sleep(0.1)
-            pass
+class CephIntegrationFilePersister(FilePersister):
```
This can removed as it is not used as of now

shtripat replied:
Done. Will remove this.

Commit:
Incorporated review comments

tendrl-spec: refactor_common_code_and_move
tendrl-bug-id: Tendrl/ceph_integration#59

Signed-off-by: Shubhendu <shtripat@redhat.com>

r0h4n reviewed (changes requested):
Fix Travis unit tests, rest LGTM
